### DISCUSSION TITLE
User Name Policy & Content Filtering

### DISCUSSION THREAD

**Discussion by sempervictus** [dani-garcia/vaultwarden#4343] [Ideas]
In a multitenant environment, it doesn't take very long for a bunch of people with the same first name to appear in the topology. This can create confusion which isn't great in credential managers.
Secondly, users can enter profanity which isn't filtered and i have not dug in to see the sanitization of inputs performed before storage/presentation to other users (and their clients) far as security concerns go.

Proposing a set of control knobs for the admin to permit enforcement of full names (first and last) and other formatting attributes along with optional content filters/validators in said policy supplied by the administrator of the instance.

**Comment by BlackDex** [comment]
I think that this is a bit out of scope for this project in my opinion. It would take a lot of knobs and tweaks to make something like this to work in a good way.

In my experience is hard to filter all and not hit valid inputs, names are to different and strange sometimes.

If you want something like this i would suggest to use something like LDAP, or, once we have it merged SSO.

Not sure if those tools provide something like that, but it could give you some control.

**Reply by sempervictus** [reply]
The invite-based account creation is great for collaborating w/ external contractors on projects where credentials have to be shared and who dont exist in internal (ADDS or similar) authentication databases or necessarily SSO setups for the entity hosting VW. Cant argue re content filtering, have made part of my living evading those in various ALGs/NIPS :smile: - could we at least have `firstname` and `lastname` fields enforcable for entry so as to "lead the horse to water" when users input their information?

**Reply by BlackDex** [reply]
There are no first add last name fields, just a single name field.
And that will not help, some people have 2 separate first name parts, some lastname, and there are also the lastname prefixes.

So, no way to filter this correctly.

**Reply by sempervictus** [reply]
Thanks, guessing that adding fields would not be great for upstream compat either. Will just have to keep an eye on the email field
